content/en/blog/falco-driverkit-debian-docker.md

```diff
@@ -0,0 +1,516 @@
+---
+title: Falco Driverkit with Docker on Debian
+linktitle: Falco Driverkit with Docker on Debian
+description: Learn how to use build your own Falco Drivers for Debian
+date: 2022-08-16
+author: Vicente J. Jiménez Miras
+slug: falco-driverkit-debian-docker
+---
+
+We use different technologies on a daily basis. Tools like Vagrant, Terraform, Ansible, and many more allow us to create and destroy digital resources in a matter of minutes, if not seconds. However, if you keep changing your running environment, you might also need to calibrate your workloads to these new changes. This is especially true when you deploy applications tightly dependent on the operating system.
+
+In other words, every time you deploy an application like Falco there's a chance that you need to compile a new module or eBPF probe to get along with the current underlying kernel. This is the first of a series of posts where you will learn some interesting techniques related to how Falco generates the much needed driver and how you can make it available for your deployments.
+
+## Falco on Docker
+
+There are many ways to run Falco: as a service, as a local container, as a Pod in Kubernetes, etc. Either way, if what we want to do is use Falco to detect threats based on syscalls, we will need a driver that has been compiled for the specific kernel running on the machine, be it a physical machine, a virtual one, or a Kubernetes node in the cloud.
+
+### Launching Falco as a container
+
+The Falco image embeds a script, `/usr/bin/falco-driver-loader`, that will automatically try to find and download a kernel module or an eBPF probe. If that wasn't possible, it might try to compile it inside the container itself. We will learn a bit more about this process and how to control it.
+
+Here is the output of a fresh instance of `falco` running on our local docker service:
+
+```shell
+$ docker run -it --privileged \
```

Comment:
maybe you could add the `--rm` argument as the image is not suppose to be re-used

Reply:
The idea is good, but I didn't add it to leave the possibility of reviewing the logs.

Reply:
makes sense